pkg/replicaset/replicaset.go

```diff
@@ -0,0 +1,446 @@
+package replicaset
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// Builder provides struct for replicaset object containing connection to the cluster and the replicaset definitions.
+type Builder struct {
+	// Replicaset definition. Used to create a replicaset object.
+	Definition *appsv1.ReplicaSet
+	// Created replicaset object.
+	Object *appsv1.ReplicaSet
+	// Used in functions that define or mutate replicaset definition. errorMsg is processed before the replicaset
+	// object is created.
+	errorMsg  string
+	apiClient goclient.Client
+}
+
+// AdditionalOptions additional options for replicaset object.
+type AdditionalOptions func(builder *Builder) (*Builder, error)
+
+var retryInterval = time.Second * 3
+
+// NewBuilder creates a new instance of Builder.
+func NewBuilder(
+	apiClient *clients.Settings, name, nsname string) *Builder {
+	glog.V(100).Infof(
+		"Initializing new replicaset structure with the following params: "+
+			"name: %s, namespace: %s", name, nsname)
+
```

Comment:
Now you can also add following validation:

	if apiClient == nil {
		glog.V(100).Infof("The apiClient is empty")

		return nil, fmt.Errorf("replicaSet 'apiClient' cannot be empty")
	}

Reply:
I think labels are also minimum req for replicaSet. Please add labels to builder as well.